From 9c56dfa4ed80933eb8779d1161f6ebf2f33cb7d5 Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Tue, 4 Sep 2012 14:10:40 -0500
Subject: [PATCH 3/4] check chdir(/) return value

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 src/lxc/lxccontainer.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lxc/lxccontainer.c b/src/lxc/lxccontainer.c
index e88460b..840679b 100644
--- a/src/lxc/lxccontainer.c
+++ b/src/lxc/lxccontainer.c
@@ -320,7 +320,10 @@ static bool lxcapi_start(struct lxc_container *c, int useinit, char ** argv)
 		if (pid != 0)
 			return wait_on_daemonized_start(c);
 		/* like daemon(), chdir to / and redirect 0,1,2 to /dev/null */
-		chdir("/");
+		if (chdir("/")) {
+			SYSERROR("Error chdir()ing to /.");
+			return false;
+		}
 		close(0);
 		close(1);
 		close(2);
-- 
1.7.10.4

