From a02264fb1ef8abab2f7aff2d347638e6e831156e Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Tue, 11 Dec 2012 11:39:26 -0600
Subject: [PATCH 2/3] README: fix typo in example script

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 README |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: lxc/README
===================================================================
--- lxc.orig/README	2012-12-11 12:41:03.000000000 -0600
+++ lxc/README	2012-12-11 12:43:02.371880877 -0600
@@ -69,7 +69,7 @@
 whitelist
 EOF
 for i in `seq 0 300`; do
-	echo $i >> secomp.full
+	echo $i >> seccomp.full
 done
 for i in `seq 1024 1079`; do
 	echo $i >> seccomp.full
