From da8c298534b0854aab9be51b5deba72649dc34e2 Mon Sep 17 00:00:00 2001
From: Christian Brauner <brauner@kernel.org>
Date: Wed, 17 Aug 2022 09:46:14 +0200
Subject: [PATCH 32/45] mount_utils: remove conf.h include

Signed-off-by: Christian Brauner (Microsoft) <christian.brauner@ubuntu.com>
---
 src/lxc/mount_utils.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/lxc/mount_utils.c b/src/lxc/mount_utils.c
index 5763afafc..123bbda77 100644
--- a/src/lxc/mount_utils.c
+++ b/src/lxc/mount_utils.c
@@ -10,7 +10,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
-#include "conf.h"
 #include "file_utils.h"
 #include "log.h"
 #include "macro.h"
-- 
2.37.2

