From 20cc784560124961e7dd173a690d70b6b6f0735e Mon Sep 17 00:00:00 2001
From: "Neil.wrz" <wangrunze13@huawei.com>
Date: Thu, 1 Sep 2022 02:13:03 -0700
Subject: [PATCH 36/45] fix error message when use tools with -? option

Signed-off-by: Neil.wrz <wangrunze13@huawei.com>
---
 src/lxc/tools/arguments.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/tools/arguments.h b/src/lxc/tools/arguments.h
index 0f0875079..92510ecbc 100644
--- a/src/lxc/tools/arguments.h
+++ b/src/lxc/tools/arguments.h
@@ -137,6 +137,7 @@ struct lxc_arguments {
 #define LXC_COMMON_OPTIONS                                                     \
 	    { "name",        required_argument, 0, 'n'         },              \
 	    { "help",        no_argument,       0, 'h'         },              \
+	    { "help",        no_argument,       0, '?'         },              \
 	    { "usage",       no_argument,       0, OPT_USAGE   },              \
 	    { "version",     no_argument,       0, OPT_VERSION },              \
 	    { "quiet",       no_argument,       0, 'q'         },              \
-- 
2.37.2

