Author: Jamie Strandboge <jamie@canonical.com>
Description: updates for AppArmor signal and ptrace mediation
Bug-Ubuntu: https://launchpad.net/bugs/1298611

Index: lxc-1.0.2/config/apparmor/abstractions/container-base
===================================================================
--- lxc-1.0.2.orig/config/apparmor/abstractions/container-base	2014-03-27 14:10:32.000000000 -0500
+++ lxc-1.0.2/config/apparmor/abstractions/container-base	2014-04-03 07:09:11.281166821 -0500
@@ -3,6 +3,8 @@
   file,
   umount,
   dbus,
+  signal,
+  ptrace,
 
   # ignore DENIED message on / remount
   deny mount options=(ro, remount) -> /,
Index: lxc-1.0.2/config/apparmor/abstractions/start-container
===================================================================
--- lxc-1.0.2.orig/config/apparmor/abstractions/start-container	2014-03-27 14:10:32.000000000 -0500
+++ lxc-1.0.2/config/apparmor/abstractions/start-container	2014-04-03 07:09:18.413331267 -0500
@@ -2,6 +2,8 @@
   capability,
   file,
   dbus,
+  signal,
+  ptrace,
 
   # currently blocked by apparmor bug
   mount -> /usr/lib/*/lxc/{**,},
